### ISSUE TITLE
Interactive quiz for lectures

### ISSUE THREAD

**Issue by danm0nster** [nielsthl/QaDiL#6]
It would be nice to have a special kind of quiz, that can be used in a live situation, such as a lecture, and where the results would be collected, and could be presented by the lecturer. I envisage something akin to Mentimeter, Poll Everywhere, Swipe, Questup. etc. Perhaps there is an open source JS tool that can be embedded?

**Comment by nielsthl** [comment]
Definitely. Here I think a very simple server needs to be set up. Correct me if I am wrong, but I do not see how the instructor can handle quizzes in a classroom situation without going through a dedicated server i.e., he needs to start and stop them and display graphs of the answers. Also, the students need to communicate with a server through their smartphones. Then again, perhaps we could be creative and bypass the server option.

**Comment by nielsthl** [comment]
This could be a life server not wanting to code and host. Thanks.
![strawpoll](https://user-images.githubusercontent.com/23010769/131221842-917ea900-2c63-4bfa-b2d7-938431188700.png)

Unfortunately I have been unable to `iframe` https://polleverywhere.com and https://menti.com (AU subscribes to the latter). These two polling system support LaTeX. It is not clear to me if `StrawPoll` supports mathematical notation in terms of LaTeX. Would be great if this was the case.

**Comment by anton-christensen** [comment]
An alternative to poll systems that don't support latex is making your own frontend with a poll system that has an API for answering questions. Then you could render questions however you want.
I might suggest socrative.com because I remember a friend making a terminal app answering quizzes (https://github.com/Hense94/socrative-cli) but I think he just reverse engineered their internal API. Doing the same might very well work for you, but it would be nice to have a service that willingly publishes their API.

**Comment by nielsthl** [comment]
It seems that https://mentimeter.com is can be embedded (see https://help.mentimeter.com/en/articles/410896-embed-your-presentation-on-a-webpage):
![menti](https://user-images.githubusercontent.com/23010769/131229573-ccfd0999-3ffa-46d3-aa7f-b5383a298e7c.png)

The truly sad state of affairs is that LaTeX only works in the question and not  in the answers. This is a mystery to me. Surely https://polleverywhere.com must be better suited for science. Just need the API.

**Comment by nielsthl** [comment]
It seems that `mentimeter` can be embedded rather easily in `QaDiL` by simply defining the macros
```
\newcommand{\mentimeter}[1]{<div style='position: relative; padding-bottom: 56.25\%; padding-top: 35px; height: 0; overflow: hidden;'><iframe sandbox='allow-scripts allow-same-origin allow-presentation' allowfullscreen='true' allowtransparency='true' frameborder='0' height='315' src='https://www.mentimeter.com/embed/#1' style='position: absolute; top: 0; left: 0; width: 100\%; height: 100\%;' width='420'></iframe></div>}

\newcommand{\menti}[1]{\begin{hideinbutton}{Mentimeter}
\mentimeter{#1}
\end{hideinbutton}
}
```
Then the button can be opened in the lecture and live voting can be presented on the web page. The input for `\menti` is simply the id for the quiz, like
```
\menti{7f433fb3f31fd6da8e9d9fd796d4db94/09dd8d3fe096}
```
![menti](https://user-images.githubusercontent.com/23010769/131397893-da7688e4-fb62-495e-bec4-fd3689e19dae.png)

**Comment by danm0nster** [comment]
This is very useful, @nielsthl ! Thanks! 

It seems Poll Everywhere allows for mathematics in the answers as well (see [here](https://support.polleverywhere.com/hc/en-us/articles/1260801547070-Math-equations-and-science-formulas)), and you can embed with `<iframe>` similarly to Mentimeter (see [here](https://support.polleverywhere.com/hc/en-us/articles/1260801821849-Embed-an-activity)). I don't have a Poll Everywhere account, so I haven't tried it.
